### ISSUE TITLE
A guard that never STARTED is scored as a block — the sweep needs a control probe, not a better path resolver

### ISSUE THREAD

**Issue by zernie** [zernie/agent-lint#213]
## The defect, in one line

`blocked` is decided by exit code 2. An interpreter that cannot start its script **also** exits 2. So a guard that never ran is indistinguishable from a guard that denied, and the sweep reports a perfect score for a program that did nothing.

```ts
// src/guardrail-check.ts
/** True iff the hook denied the call (exit 2 / decision:block / permissionDecision:deny). */
readonly blocked: boolean;
```

## Why this is one bug and not five

`experimental_verifyPluginGuards` guards against it with a **preflight**: resolve the hook's script and report `unresolved` when it is not on disk. That preflight is a *proxy* for "did the guard actually run", and review has now found five separate ways to defeat the proxy, each fixed individually:

| form | example |
| --- | --- |
| wrapper hides the interpreter | `env python3 guard` |
| wrong execution cwd | sweeping a repo other than `process.cwd()` |
| command-local assignment | `GUARD=missing.py; python3 "$GUARD"` |
| a dominating `cd` | `cd hooks && python3 guard.py` |
| command substitution | `result=$(python3 missing.py)` |

The first two are fixed on `claude/verify-plugin-guards`; the last three are open.

**Resolving harder will not converge.** Extracting a script path from an arbitrary shell command is the same undecidable problem `bash-effects.ts` already documents for itself — `eval`, a `$VAR` head, `sh -c` and `base64` normalize to `null` *by construction*, precisely because a sound answer does not exist. Every round of patching buys one more spelling and leaves the class open.

## The fix that closes the class

Feed the hook one **benign control command** it must ALLOW, alongside the disaster battery.

- control allowed, disasters blocked → the guard is real; score it.
- control **also blocked** → the program is not blocking, it is failing to start. Report `unmeasurable` with that reason, never a score.

This makes the whole family unrepresentable, including spellings nobody has found yet, because it stops inferring "did it run" from a path and starts observing it from behaviour. It costs one extra `runHook` per hook.

The pattern is already used in this repository — `src/subagent-delivery.test.ts` establishes delivery with an in-run control:

> the parent's identical command in the same run is the in-run control; ground truth on disk, not the trace

## Notes

- The control must be something a *correct* guard allows. A benign `echo`-shaped command is the obvious candidate; the disaster catalog's own "allows" cases are already exercised as false-positive checks in `hook-oss-comparison.test.ts`, so the vocabulary exists.
- A guard legitimately narrow enough to block the control is possible in principle. That case is still better reported as `unmeasurable` than as a score, per this feature's own rule that a missing RUN must never read as a finding.
- Until this lands, the three open spellings are named as a known remainder in the `verify-plugin-guards.ts` module header rather than left silent.

**Comment by zernie** [comment]
Sixth spelling, found on `e306a03` roughly an hour after the fifth — and it is the strongest evidence yet that the count is the wrong thing to chase.

**`python3 -E missing.py`.** `-E` means "ignore `PYTHON*` environment variables" and consumes no operand, but it sits in the shared `NON_PATH_FLAGS` table, so the walk treats it like an eval-style flag, skips the script operand, and emits no file reference. The preflight therefore passes, Python exits 2 for the absent script, and the sweep can read that startup failure as a block across the whole battery.

**The honest counter-argument, stated because it is a real one.** Unlike `$VAR`, `eval` and `$(…)`, this one *is* bounded: which flags consume program text is a finite, decidable fact per interpreter, and the reviewer's proposed fix — make the flag table interpreter-specific instead of global — is correct and closes this spelling properly. So "undecidable" is not why this is routed here.

Why it is routed here anyway:

- The wrapper table was finite too, and being finite did not stop `exec` from being missing from it, nor `sudo g\it push` from surviving it.
- A per-interpreter flag table is a **hand-maintained list**, which is the exact shape this repository already has a rule against: a fact re-stated by hand in a second place falls behind. `python3`, `node`, `ruby`, `perl`, `sh` and `bash` each have their own; the next one added to the table is one nobody will notice is missing.
- Most importantly, closing it changes nothing about the failure mode. Even with a perfect per-interpreter table, `cd hooks && …`, `GUARD=x; … "$GUARD"` and `result=$(…)` still produce the same false score, because the defect is not in any table — it is that **exit 2 is read as a deny without ever establishing that the program ran.**

Six spellings in two days, each found after the previous was fixed, is the measurement this issue exists to record. The control probe does not need to know what `-E` means.

---

One note for whoever implements this: the `KNOWN REMAINDER` block in `verify-plugin-guards.ts` currently says "defeated that proxy **five times**". That count is now wrong and will keep going wrong — it should name the class and drop the number, per this repository's own rule that a number in prose goes stale silently. Worth folding into the same change rather than spending a CI round on the count alone.

---
_Generated by [Claude Code](https://claude.ai/code)_
